api/v1alpha1/hosted_controlplane.go

```diff
@@ -32,10 +32,15 @@ type HostedControlPlaneSpec struct {
 	ServiceCIDR  string                      `json:"serviceCIDR"`
 	PodCIDR      string                      `json:"podCIDR"`
 	MachineCIDR  string                      `json:"machineCIDR"`
-	SSHKey       corev1.LocalObjectReference `json:"sshKey"`
-	InfraID      string                      `json:"infraID"`
-	Platform     PlatformSpec                `json:"platform"`
-	DNS          DNSSpec                     `json:"dns"`
+	// NetworkType specifies the SDN provider used for cluster networking.
+	// +kubebuilder:default:="OpenShiftSDN"
+	// +unionDiscriminator
```

Comment:
not sure that this applies here

Reply:
cool I am by no means an expert on unionDiscriminator so appreciate the insight